File: frontend/src/services/marketplaceService.js
import apiClient from './api';

export const getPlugins = () => apiClient.get('/suite/marketplace/plugins');
export const installPlugin = (dto) => apiClient.post('/suite/marketplace/plugins', dto);
export const togglePlugin = (id, enabled) => apiClient.put(`/suite/marketplace/plugins/${id}/toggle`, { enabled });
export const uninstallPlugin = (id) => apiClient.delete(`/suite/marketplace/plugins/${id}`);

export const uploadPluginBundle = async (file) => {
  const formData = new FormData();
  formData.append('file', file);
  
  // Custom fetch for multipart file upload
  const token = localStorage.getItem('token');
  const tenant = localStorage.getItem('tenant_subdomain') || 'awais';
  
  const headers = {};
  if (token) headers['Authorization'] = `Bearer ${token}`;
  if (tenant) headers['X-Tenant'] = tenant;

  const response = await fetch('/api/v1/suite/marketplace/plugins/upload', {
    method: 'POST',
    headers,
    body: formData
  });

  const data = await response.json();
  if (!response.ok) {
    throw new Error(data?.message || 'Failed to upload plugin bundle');
  }
  return data;
};

File: backend/src/main/java/com/awais/hr/module/marketplace/service/MarketplaceService.java
package com.awais.hr.module.marketplace.service;

import java.io.File;
import java.util.List;
import java.util.Map;

public interface MarketplaceService {
    List<Map<String, Object>> getPlugins();
    Map<String, Object> installPlugin(Map<String, Object> body);
    Map<String, Object> togglePlugin(String id, boolean enabled);
    Map<String, Object> uninstallPlugin(String id);
    Map<String, Object> uploadAndInstallPluginBundle(File tempFile, String filename);
}
